libcontainer/intelrdt/intelrdt.go

```diff
@@ -589,6 +600,28 @@ func (m *Manager) GetStats() (*Stats, error) {
 	return stats, nil
 }
 
+func combineSchemas(l3CacheSchema, memBwSchema string) string {
+	// If both l3CacheSchema and memBwSchema are set and
+	// l3CacheSchema contains a line beginning with "MB:", the
+	// value written to schemata file MUST be the non-"MB:"
+	// line(s) from l3CacheSchema and the line from memBWSchema.
```

Comment:
So, can you elaborate on that requirement? In particular, why we allow a user to include `MB:` lines into `l3CacheSchema`? Would it be better to offload this requirement to a user of runc (whoever creates the OCI spec)? I mean, runc is a low level tool and should, theoretically, be as simple as possible.

My other question is -- what happens if we write both lines as they are?

Reply:
One other thought. Let's say that we have
```
memBwSchema         = "MB:0=70;1=20"
l3CacheSchema       = "MB:0=80;1=10\nL3:0=f0;1=f"
```

Does it make a difference for the kernel, if we write

```
MB:0=80;1=10
L3:0=f0;1=f
MB:0=70;1=20
```

or just

```
L3:0=f0;1=f
MB:0=70;1=20
```
?

(I mean, except the additional line parsing done in the kernel)

Reply:
To answer your other question: I tried this out now and having two MB values for the same domain in the same `write` won't work:

```
[root@wolfpass ipuustin]# uname -r
6.0.10-100.fc35.x86_64
[root@wolfpass ipuustin]# cat schemata-1.txt
MB:0=80;1=90
L3:0=ff;1=ff
[root@wolfpass ipuustin]# cat schemata-2.txt
MB:0=80;1=90
L3:0=ff;1=ff
MB:0=60;1=70
[root@wolfpass ipuustin]# cat schemata-1.txt > /sys/fs/resctrl/foo/schemata
[root@wolfpass ipuustin]# cat /sys/fs/resctrl/foo/schemata
    MB:0= 80;1= 90
    L3:0=0ff;1=0ff
[root@wolfpass ipuustin]# cat schemata-2.txt > /sys/fs/resctrl/foo/schemata
cat: write error: Invalid argument
[root@wolfpass ipuustin]# cat /sys/fs/resctrl/info/last_cmd_status
Duplicate domain 0
```

So we know that there is nobody trying to set overlapping `MB:` values in both `memBwSchema` and `l3CacheSchema` because that would now trigger an error in runc. There might be someone setting non-overlapping `MB:` values and not getting those filtered out however.